Add test for fixed parsing of goto.

DemiMarie opened the pull request:
Also update test for changes to the table library.

DemiMarie committed:
Add test for fixed parsing of goto.

Also update test for changes to the table library.

DemiMarie committed:
Added many FFI tests to test runner and fix C lib loading

This adds many FFI tests to the test runner.  It also fixes loading
of C libraries by the test suite, by loading them before the global
environment is frozen.

DemiMarie committed:
Add more tests to the test suite

DemiMarie commented:
Added ~200 new test to the test runner.

lukego commented:
Regression?

```
$ git log --oneline -1
fda2015 Add more tests to the test suite

$ ../../luajit/src/luajit test.lua
../../luajit/src/luajit: test.lua:10: module 'ctest' not found:
        no field package.preload['ctest']
        no file './ctest.lua'
        no file '/usr/local/share/luajit-2.1.0-beta2/ctest.lua'
        no file '/usr/local/share/lua/5.1/ctest.lua'
        no file '/usr/local/share/lua/5.1/ctest/init.lua'
        no file './ctest.so'
        no file '/usr/local/lib/lua/5.1/ctest.so'
        no file '/usr/local/lib/lua/5.1/loadall.so'
stack traceback:
        [C]: in function 'require'
        test.lua:10: in main chunk
        [C]: at 0x00404a70
```

sindrom91 commented:
I don't think this is a regression. These commits add ffi tests to the runner, so you need to compile libraries from test/src. Also, make sure that your libraries are in a place where they can be found (you might need to export appropriate LUA_CPATH).

lukego commented:
To me it looks like a regression because prior to these changes the instructions in `test/README.md` worked for me but now they do not. Excerpt:

```
## Running the test suite ##

To run the default test suite, run `test.lua` using the Lua interpreter you
wish to test, for example:

    $ ~/luajit-2.0/src/luajit test.lua
```

sindrom91 commented:
Oh, I see what you're saying. I didn't look at it that way. :)

DemiMarie committed:
Fix instructions in README.md.  Add more tests to runner.

DemiMarie committed:
Add more FFI tests to the test suite

DemiMarie committed:
FFI JIT call tests

DemiMarie commented:
There are some tests still failing.  I believe that they indicate genuine bugs in either LuaJIT or the test suite.

I fixed the instructions in `README.md`.

DemiMarie committed:
Add new FFI tests

This adds many FFI tests.  It also fixes tests for the contents
of the string and math libraries.
